repaygithub/cactus
# CACTUS-303 :: DataGrid Card Sorting

Ticket: https://repayonline.atlassian.net/browse/CACTUS-303

Alright, I've got a few things going on here so I'll break it down:

- I added a new prop `cardBreakpoint` to `DataGrid` which works the same as the `cardBreakpoint` on the table. It defaults to `tiny`. Previously, the `DataGrid` switched to card view at `small`, so this fixes the weird-ness seen in the screenshot on the ticket.
- This adds a new variant to the `MenuButton` -- I did this because we need it for the card view sorting dropdowns, and the design was just too similar to the `MenuButton` to justify creating an ad-hoc component inside the `DataGrid`.
- I'm also adding the `DataGrid` to the Accounts page in the mock-ebpp app and using that for integration tests.
- I added a CLI arg (`-f`, `--fixture`) in the `run-ete-tests` script to allow you to pass a fixture to TestCafe to filter on for the mock-ebpp tests.

### Note
I want to point out some weird behavior of the `MenuButton` that I noticed and get your guys' opinions on it: In the card view, in the "Order" dropdown, you can see that the restricted width on the dropdown causes the options (Ascending, Descending) to be broken up in the middle of the word. It's because the text in the trigger button is shorter than the text in the options.I don't love that, and I'd like to suggest that we give the dropdown a minimum width and come up with a value that we are comfortable with the dropdown growing to based on the size of the trigger. For example, we could say something like:
```
minWidth: targetRect.width,
maxWidth: targetRect.width + 20
```
But I'm curious what your thoughts are on that issue and if anyone has an opinion on it.


### Testing
- Set the `cardBreakpoint` to `small` in the DataGrid storybook.
- Set the viewport to `tiny` and test the sorting features in the card view
- Set the viewport to `small` and test the sorting features. Notice that the elements above/below the grid move around a bit when switching between these two breakpoints
- Do some playing around with other features both in card view and table view to double check that this didn't break anything else.
